testutil-server/src/main/java/io/spine/server/integration/AcknowledgementsVerifier.java

```diff
@@ -0,0 +1,612 @@
+    public AcknowledgementsVerifier withRejections(RejectionClass type) {
+        AcknowledgementsVerifier rejectedType = ackedWithRejections(type);
+        return this.and(rejectedType);
+    }
+
+    /**
+     * Creates a new verifier adding a check to contain a {@link Rejection rejection} of a
+     * type specified by a {@link RejectionClass rejection class} specified amount of times.
+     *
+     * @param expectedCount an amount of rejection that are expected in Bounded Context
+     * @param type          rejection type in a form of message class
+     * @return a new {@link AcknowledgementsVerifier} instance
+     */
+    public AcknowledgementsVerifier withRejections(int expectedCount, Class<? extends Message> type) {
+        AcknowledgementsVerifier rejectedType = ackedWithRejections(expectedCount, type);
+        return this.and(rejectedType);
+    }
+
+    /**
+     * Creates a new verifier adding a check to contain a {@link Rejection rejection} of a
+     * type specified by a {@link RejectionClass rejection class} specified amount of times.
+     *
+     * @param expectedCount an amount of rejection that are expected in Bounded Context
+     * @param type          rejection type in a form of {@link RejectionClass RejectionClass}
+     * @return a new {@link AcknowledgementsVerifier} instance
+     */
+    public AcknowledgementsVerifier withRejections(int expectedCount, RejectionClass type) {
+        AcknowledgementsVerifier rejectedType = ackedWithRejections(expectedCount, type);
+        return this.and(rejectedType);
+    }
+
+    /**
+     * Creates a new verifier adding a check to contain a provided domain rejection.
+     *
+     * @param type      a type of a domain rejection specified by a {@link RejectionClass}
+     * @param predicate a predicate filtering domain rejections
+     * @param <T>       a domain rejection type
+     * @return a new {@link AcknowledgementsVerifier} instance
+     */
+    public <T extends Message> AcknowledgementsVerifier
+    withRejections(Class<T> type, RejectionPredicate<T> predicate) {
+        AcknowledgementsVerifier oneRejection = ackedWithRejections(type, predicate);
+        return this.and(oneRejection);
+    }
+
+    /**
+     * Creates a new verifier adding a check to contain a provided domain rejection.
+     *
+     * @param expectedCount an amount of rejection that are expected in Bounded Context
+     * @param type          a type of a domain rejection specified by a {@link RejectionClass}
+     * @param predicate     a predicate filtering domain rejections
+     * @param <T>           a domain rejection type
+     * @return a new {@link AcknowledgementsVerifier} instance
+     */
+    public <T extends Message> AcknowledgementsVerifier
+    withRejections(int expectedCount, Class<T> type, RejectionPredicate<T> predicate) {
+        AcknowledgementsVerifier oneRejection = ackedWithRejections(expectedCount, type, predicate);
+        return this.and(oneRejection);
+    }
+}
```

Comment:
This class seems to be a bit bigger than seems manageable. Can we please extract some of its bits as package-private classes under the same package?